Protos/V1/agent_common.proto

```diff
@@ -70,13 +73,144 @@ message Result {
+}
+
+/**
+* Request to create tasks.
+*/
+message SubmitTasksRequest {
+  message TaskCreation {
+    repeated string expected_output_keys = 1; /** Unique ID of the results that will be produced by the task. Results are created using ResultsService. */
+    repeated string data_dependencies = 2; /** Unique ID of the results that will be used as datadependencies. Results are created using ResultsService. */
+    string payload_id = 3; /** Unique ID of the result that will be used as payload. Results are created using ResultsService. */
+    TaskOptions task_options = 4; /** Optionnal task options. */
+  }
+
+  string session_id = 1; /** The session ID. */
+  TaskOptions task_options = 2; /** The options for the tasks. Each task will have the same. Options are merged with the one from the session. */
+  repeated TaskCreation task_creations = 3; /** Task creation requests. */
+  string communication_token = 4; /** Communication token received by the worker during task processing */
+}
+
+/**
+* Response to create tasks.
+*
+* expected_output_ids and data_dependencies must be created through ResultsService.
+*
+* Remark : this may have to be enriched to a better management of errors but
+* will the client application be able to manage a missing data dependency or expected output ?
+*/
+message SubmitTasksResponse {
+  message TaskInfo {
+    string task_id = 1; /** The task ID. */
+    repeated string expected_output_ids = 2; /** The expected output IDs. A task has expected output IDs. */
+    repeated string data_dependencies = 3; /** The data dependencies IDs (inputs). A task has data dependencies. */
+    string payload_id = 4; /** Unique ID of the result that will be used as payload. Results are created implicitly. */
+  }
+
+  repeated TaskInfo task_infos = 1; /** List of task infos if submission successful, else throw gRPC exception. */
+  string communication_token = 2; /** Communication token received by the worker during task processing */
+}
+
+/*
+* Request for creating results without data
+*/
+message CreateResultsRequest {
+  /**
+  * A result to create.
+  */
+  message ResultCreate {
+    bytes data = 1; /** The actual data of the result. */
+    string name = 2; /** The result name. Given by the client. */
+  }
+  repeated ResultCreate results = 1; /** The results to create. */
+  string session_id = 2; /** The session in which create results. */
+  string communication_token = 3; /** Communication token received by the worker during task processing */
+}
+
+/*
+* Response for creating results without data
+*/
+message CreateResultsResponse {
+  repeated ResultMetaData results = 1; /** The raw results that were created. */
+  string communication_token = 2; /** Communication token received by the worker during task processing */
+}
+
+/*
+* Request for uploading results data through stream.
+* Data must be sent in multiple chunks
+*/
+message UploadResultDataRequest {
```

Comment:
With such a stream message, it would be possible to support multiple uploads in the same stream without any addition to the message. Should we allow it, or forbid it? In any case, we should be explicit about it.

Reply:
As the code is implemented, we support only one send